altapay.php

```diff
@@ -0,0 +1,2390 @@
+        $cgConf['uniqueid']     = $cart->id;
+        $cgConf['terminal']     = $terminal->remote_name;
+        $cgConf['cookie']       = isset($_SERVER['HTTP_COOKIE']) ? $_SERVER['HTTP_COOKIE'] : null;
+
+        $callback = [];
+        // Callbacks
+        $callback['callback_form']         = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackform',
+            [],
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_ok']           = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackok',
+            [],
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_fail']         = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackfail',
+            [],
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_open']         = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackopen',
+            [],
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_notification'] = $this->context->link->getModuleLink(
+            $this->name,
+            'callbacknotification',
+            [],
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $callback['callback_redirect']     = $this->context->link->getModuleLink(
+            $this->name,
+            'callbackredirect',
+            [],
+            true,
+            $this->context->language->id,
+            $this->context->shop->id
+        );
+        $customer                          = [];
+        // Customer info
+        $customer['billing_firstname'] = $this->context->customer->firstname;
+        $customer['billing_lastname']  = $this->context->customer->lastname;
+        $customer['email']             = $this->context->customer->email;
+
+        // Billing address
+        $invoice_address = new Address($this->context->cart->id_address_invoice);
+        $country         = new Country($invoice_address->id_country);
+        $state           = new State($invoice_address->id_state);
+
+        $customer['billing_address'] = $invoice_address->address1;
+        $customer['billing_city']    = $invoice_address->city;
+        $customer['billing_postal']  = $invoice_address->postcode;
+        $customer['billing_region']  = $state->iso_code;
+        $customer['billing_country'] = $country->iso_code;
+
+        // Phone
+        $invoiceAph                 = $invoice_address->phone;
+        $customer['customer_phone'] = $customer['customer_phone'] = $invoice_address->phone ?: $invoice_address->phone_mobile;
```

Comment:
Somethign went wrong when cleaning up this code, it should be:
```suggestion
        $customer['customer_phone'] = $invoice_address->phone ?: $invoice_address->phone_mobile;
```